src/tests/JIT/opt/LocalMasks/ChangeMatchUse.cs

```diff
@@ -0,0 +1,182 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+// Unit tests for the local masks optimization
+
+using System;
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.Arm;
+using System.Threading;
+using Xunit;
+
+public class AcrossAndCselToAcross
+{
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    private static void Consume<T>(T value) { }
+
+    [MethodImpl(MethodImplOptions.NoInlining)]
+    private static void Consume<T, T2>(T value, T2 value2) { }
+
+
+    // Create a mask. Use it as a mask.
+    // Conversion of mask1 will be removed.
+    [Fact]
+    public static void UseMaskAsMask()
+    {
+        for (int i = 0; i < 4; i++)
+        {
+            for (int j = 0; j < 200; j++)
+            {
+                InnerUseMaskAsMask();
+            }
+
+            Thread.Sleep(100);
+        }
+        InnerUseMaskAsMask();
+    }
```

Comment:
I would suggest having just a single `Fact` so that you can tier up all tests simultaneously -- otherwise this test is going to take longer than necessary (it doesn't take long on its own, but all of it adds up..). You could potentially even make this test run only when `ManualTestsEnabled` (if you do that then the separate facts seem ok to me).